### PR TITLE
feat: Notify endpoint and Parcel Delivery

### PR DESCRIPTION
**Pull request by sdsantos**

### CODE REVIEW

**Comment by sdsantos** [comment]
Double-check the `generateEndpointCertificate` method, cause I'm not sure that's what it should look like.

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/NotifyEndpoint.kt [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
-    // TODO: implementation
-    suspend fun notify(endpointAddress: MessageAddress) {}
+    suspend fun notify(endpointAddress: MessageAddress) {
+        endpointDao.get(endpointAddress)
+            ?.let { endpoint ->
+                context.sendBroadcast(
+                    Intent(RelaynetAndroid.ENDPOINT_NOTIFY_ACTION)
+                        .setPackage(endpoint.applicationId),
+                    RelaynetAndroid.GATEWAY_SYNC_PERMISSION
+                )
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/gateway/domain/endpoint/NotifyEndpoint.kt:24]: Just wanted to double check: This isn't broadcasting the event to many/all apps on the phone -- Instead, it's limiting the message to the endpoint's app. Right?

I think that's what happening but I wanted to check because of the word "broadcast".

Or is there any chance that third party developers can register intents for other apps? Or simply monitor broadcasts for other apps?

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/RegisterEndpoint.kt [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
     suspend fun register(applicationId: String, endpointAddress: MessageAddress) {
         localEndpointDao.insert(LocalEndpoint(endpointAddress, applicationId))
     }
+
+    suspend fun getGatewayCertificate() = localConfig.getCertificate()
+
+    suspend fun generateEndpointCertificate(endpointAddress: MessageAddress) {
+        val gatewayKeyPair = localConfig.getKeyPair()
+        issueEndpointCertificate(
+            gatewayKeyPair.public,
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/gateway/domain/endpoint/RegisterEndpoint.kt:26]: This should be the public key of the endpoint, which we need to receive as a parameter instead of the endpoint's address.

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/RegisterEndpoint.kt [hunk 3]
```diff
@@ -1,16 +1,36 @@
 package tech.relaycorp.gateway.domain.endpoint
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/gateway/domain/endpoint/RegisterEndpoint.kt:1]: I had to issue this cert as part of #72, so I think we can undo the changes to this file (the new method doesn't seem to be used anyway).

File: app/build.gradle [hunk 4]
```diff
@@ -63,6 +63,8 @@ android {
     }
     packagingOptions {
         exclude 'META-INF/*'
+        exclude 'META-INF/licenses/*'
+        exclude '**/attach_hotspot_windows.dll'
```
[the remark this answers is not in the document]
Comment (sdsantos) [reply app/build.gradle:67]: Excluding the `META-INF` ones is pretty common, but the dll isn't. Going to add a comment.

File: app/src/main/java/tech/relaycorp/gateway/domain/endpoint/NotifyEndpoint.kt (hunk 1 above)
Comment (sdsantos) [reply app/src/main/java/tech/relaycorp/gateway/domain/endpoint/NotifyEndpoint.kt:24]: Any app can subscribe to a certain broadcast, but since we're using the `setPackage` filter, only components inside that application will receive them.
